File: ZegaFinancials.Web/ClientApp/src/app/models/DataGrid/FilterExpressionModel/filter-expression-model.ts
import { Data } from "@angular/router";
import { DateTimeConditionTypes, NumericConditionTypes, StringConditionTypes } from "../../../support/enums/filter.enum";
import { DataGridColumnModel } from "../DataGridColumnModel/data-grid-column-model";

export class FilterExpressionModel {

  constructor(obj: any) {
    this.ExpressionGuid = obj?.expressionGuid ?? '';
    this.SelectedValues = obj?.selectedValues ?? [];
    this.StringConditionType = obj?.stringConditionType ?? StringConditionTypes.Contains;
    this.NumericConditionType = obj?.numericConditionType ?? NumericConditionTypes.Equals;
    this.DateTimeConditionType = obj?.dateTimeConditionType ?? DateTimeConditionTypes.WithInLast;
    this.Values = obj?.values ?? {};
    this.DateConfig = obj?.dateConfig ?? {};
  }

  ExpressionGuid: string
  SelectedValues: Array<number>;
  StringConditionType: StringConditionTypes;
  NumericConditionType: NumericConditionTypes;
  DateTimeConditionType: DateTimeConditionTypes;
  Values: FilterExpressionValues;
  DateConfig: DateConfigModel
}

interface FilterExpressionValues {
  Value1: string;
  Value2: string;
  Value3: string;
  Value4: string;
  Value5: string;
}

interface DateConfigModel {
  StartDate: Date;
  EndDate: Data;
}

export interface AdvancedFilterCommunicationModel {
  GridColumns: Array<DataGridColumnModel>
  ApplyFilter: boolean
}

File: ZegaFinancials.Nhibernate/Shared/Models/FilterExpressionModel.cs
using System;
using System.Collections.Generic;
using ZegaFinancials.Nhibernate.Shared.Enums;

namespace ZegaFinancials.Nhibernate.Shared.Models
{
    public class FilterExpressionModel
    {
        Guid _obj = Guid.NewGuid();
        public FilterExpressionModel()
        {

            DateConfig = new DateConfigModel();
            Values = new Dictionary<string, string> { { "Value1", "" }, { "Value2", "" }, { "Value3", "" }, { "Value4", "" }, { "Value5", "" } };
            StringConditionType = StringConditionTypes.Contains;
            NumericConditionType = NumericConditionTypes.Equals;
            DateTimeConditionType = DateTimeConditionTypes.WithInLast;
            SelectedValues = new List<int>();
        }

        public string ExpressionGuid => _obj.ToString().Substring(0, 8);
        public List<int> SelectedValues { get; set; }
        public Dictionary<string, string> Values { get; set; }
        public StringConditionTypes StringConditionType { get; set; }
        public NumericConditionTypes NumericConditionType { get; set; }
        public DateTimeConditionTypes DateTimeConditionType { get; set; }
        public DateConfigModel DateConfig { get; set; }
    }
}
